### DISCUSSION TITLE
Intermittent Static on all Clients when playing through librespot

### DISCUSSION THREAD

**Discussion by degan6** [snapcast/snapcast#1149] [Q&A]
I've set up a Snapcast server on a Ubuntu VM and compiled librespot and can stream Spotify to clients, but sadly, the stream is unable because of intermittent static.

I've [recorded this short video](https://youtu.be/VucZ3m_Vx0g) that demonstrates the problem. **The video has some loud static in it!**

In the video I:
- Connect Spotify to the Snapcast stream named Kitchen. 
- You can see the Windows CLI clients next to the Spotify window start getting chunks
- Right as Spotify connects to the client, some popping sounds always follow from the client's speakers
- Hit play on Spotify, and skip through some songs.
- Towards the end is worst, when the stream just turns to static.

This static is synced to all connected clients, not just my test CLI Windows clients. This is on a wired network. 

My only guess right now is something with the sample rate from the Spotify stream, but the audio sounds fine when it working, and most other issues seem to say that the audio will sound fast or slow.

I would love some help/ideas!

I'm running: 

>snapserver: Version 0.27.0, revision 54a3d862

>librespot 0.5.0-dev f037e46 (Built on 2023-08-05, Build ID: Oo5Ouviy, Profile: release)

>Ubuntu 20.04.5 LTS

Config file without comments:

```
[server]

[http]

doc_root = /usr/share/snapserver/snapweb

codec = ogg

stream = spotify:///librespot?name=Spotify_Kitchen&devicename=Kitchen&bitrate=320&volume=100&onevent=/home/de/spotify.sh&sampleformat=44100:16:2
```

**Comment by ackstorm23** [comment]
I run into this too.  Incredibly loud static coming from librespot as if snapcast is failing to set or expect the correct samplerate.,
